test/common/upstream/subset_lb_test.cc

```diff
@@ -1984,13 +2000,23 @@ TEST_F(SubsetLoadBalancerTest, EnabledScaleLocalityWeightsRounding) {
   TestLoadBalancerContext context({{"version", "1.0"}});
 
   // We expect to see a 33/66 split because 2 * 1 / 2 = 1 and 2 * 3 / 4 = 1.5 -> 2
-  EXPECT_EQ(host_set_.healthy_hosts_per_locality_->get()[1][0], lb_->chooseHost(&context));
-  EXPECT_EQ(host_set_.healthy_hosts_per_locality_->get()[0][0], lb_->chooseHost(&context));
-  EXPECT_EQ(host_set_.healthy_hosts_per_locality_->get()[1][1], lb_->chooseHost(&context));
-  EXPECT_EQ(host_set_.healthy_hosts_per_locality_->get()[1][2], lb_->chooseHost(&context));
-  EXPECT_EQ(host_set_.healthy_hosts_per_locality_->get()[0][0], lb_->chooseHost(&context));
-  EXPECT_EQ(host_set_.healthy_hosts_per_locality_->get()[1][0], lb_->chooseHost(&context));
-  EXPECT_EQ(host_set_.healthy_hosts_per_locality_->get()[1][1], lb_->chooseHost(&context));
+  std::vector<size_t> counts({0, 0});
+  const size_t iters = WRSQScheduler<int>::accuracy();
```

Comment:
I think that the test below will fail if you change this line to:
`const size_t iters = 10 * WRSQScheduler<int>::accuracy();`

The reason behind this is that with sum_weights = 3 and accuracy of 10000, you accumulate 1 error every 10,000 random inputs processed.  The original mod-based random implementation doesn't accumuluate errors over the accuracy period.